Update nexus_upload action to allow for flexible nexus mount paths and enable logging

chrisgibbs opened the pull request:
Addresses issue: https://github.com/fastlane/fastlane/issues/1303 by allowing the Nexus mount path to be provided as part of the `endpoint` parameter.

**Note**: This would be a breaking change to anyone using the `nexus_upload` action previously. It will require them to explicitly add `/nexus` to the end of their `endpoint` parameter. I believe this is the best course for this fastlane action but if others feel strongly the other way I can modify the change to include an optional `nexus_path` parameter that would default to `/nexus`.

Also addresses issue https://github.com/fastlane/fastlane/issues/1304 by enabling logging based on the `verbose` parameter.

chrisgibbs commented:
@xfreebird as the original author of the `nexus_upload` action, just wanted to make you aware of this PR.

xfreebird commented:
@chrisgibbs Thanks ! 
Regarding of mount path, I wasn't aware of that. But would prefer not to break the compatibility.
I think the best approach would be to add a new optional option named ```mount_path```, that by default would be ```nexus```. 

In this way the plugin wont break for current users, and you'll fix your issue.

What do you think ?

mfurtak commented:
@chrisgibbs Thanks for the ticket and contribution! :+1: 

I agree with @xfreebird's proposal here around making sure existing users of the action keep functioning as before.

chrisgibbs commented:
@xfreebird @mfurtak Sounds good. I'll make the necessary changes.

Commit:
Add mount_path parameter to nexus_upload action and update associated tests

Fix nexus_upload url spec to provide a default mount_path parameter

Add shellescape to mount_path parameter

chrisgibbs commented:
Ok, should be good to go now.

mfurtak commented on lib/fastlane/actions/nexus_upload.rb:
```diff
@@ -10,11 +10,11 @@ def self.run(params)
 
       def self.upload_url(params)
-        "#{params[:endpoint].shellescape}/nexus/service/local/artifact/maven/content"
+        "#{params[:endpoint].shellescape}#{params[:mount_path]}/service/local/artifact/maven/content"
```
Let's `.shellescape` this new param as well.

mfurtak commented:
Looks great to me - just left one small comment. How's it look to you @xfreebird?

After that, if you can squash down to one commit, I think we can get this in! :+1:

xfreebird commented:
@mfurtak It looks good, thanks !

mfurtak commented:
Merged as #1325
